contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2260,6 +2403,133 @@ class tsqlBuilder : public tsqlCommonMutator
+
+		if (does_object_name_need_delimiter(ctx->tablename))
+			rewritten_query_fragment.emplace(std::make_pair(ctx->tablename->start->getStartIndex(), std::make_pair(::getFullText(ctx->tablename), delimit_identifier(ctx->tablename))));
+
+		// qualified identifier doesn't need delimiter
+		if (ctx->DOT().empty() && does_object_name_need_delimiter(ctx->column_name))
+			rewritten_query_fragment.emplace(std::make_pair(ctx->column_name->start->getStartIndex(), std::make_pair(::getFullText(ctx->column_name), delimit_identifier(ctx->column_name))));
+	}
+
+	/* For Spatial Handling See Comment in tsqlCommonMutator */
+	void exitFunc_call_expr(TSqlParser::Func_call_exprContext *ctx) override
+	{
+		if(ctx != NULL && !ctx->DOT().empty())
+		{
+			std::vector<TSqlParser::Method_callContext *> method_calls = ctx->method_call();
+			for (size_t i=0; i<method_calls.size(); ++i)
+			{
+				TSqlParser::Method_callContext *method = method_calls[i];
+				/* rewriting the query in case of Geospatial function Call -> function_call.method_call */
+				if(method->spatial_methods())
+				{
+					size_t ind;
+					if (i == 0) ind = ctx->function_call()->stop->getStopIndex();
+					else ind = method_calls[i-1]->stop->getStopIndex();
+					rewrite_geospatial_query_helper(ctx, method, ind);
+				}
+			}
+		}
+	}
+
+	/* For Spatial Handling See Comment in tsqlCommonMutator */
+	void exitLocal_id_expr(TSqlParser::Local_id_exprContext *ctx) override
+	{
+		if(ctx != NULL && !ctx->DOT().empty())
+		{
+			std::vector<TSqlParser::Method_callContext *> method_calls = ctx->method_call();
+			for (size_t i=0; i<method_calls.size(); ++i)
+			{
+				TSqlParser::Method_callContext *method = method_calls[i];
+				/* rewriting the query in case of Geospatial function Call -> local_id.method_call */
+				if(method->spatial_methods())
+				{
+					size_t ind;
+					if (i == 0) ind = ctx->local_id()->stop->getStopIndex();
+					else ind = method_calls[i-1]->stop->getStopIndex();
+					rewrite_geospatial_query_helper(ctx, method, ind);
+				}
+			}
+		}
+	}
+
+	/* For Spatial Handling See Comment in tsqlCommonMutator */
+	void exitBracket_expr(TSqlParser::Bracket_exprContext *ctx) override
+	{
+		if(ctx != NULL && !ctx->DOT().empty())
+		{
+			std::vector<TSqlParser::Method_callContext *> method_calls = ctx->method_call();
+			for (size_t i=0; i<method_calls.size(); ++i)
+			{
+				TSqlParser::Method_callContext *method = method_calls[i];
+				/* rewriting the query in case of Geospatial function Call -> LR_BRACKET expression RR_BRACKET.method_call */
+				if(method->spatial_methods())
+				{
+					size_t ind;
+					std::string context = ::getFullText(ctx);
+					size_t spaces = 0;
+					for (size_t x = ctx->expression()->stop->getStopIndex() + 1 - ctx->start->getStartIndex(); x <= ctx->stop->getStopIndex() - ctx->start->getStartIndex(); x++)
+					{
+						if(context[x] == ' ') spaces++;
+						else if(context[x] == ')') break;
+					}
```

Comment:
Yes, It is at the end of expression. 
Because in this rule we do not have access to that index directly so we have to iterate over and we can encounter spaces, whereas in other rules we can access then directly.